protocol/v2/ssv/validator/committee.go

```diff
@@ -150,10 +151,18 @@ func (c *Committee) PushToQueue(slot phase0.Slot, dec *queue.DecodedSSVMessage)
 	}
 }
 
-func removeIndex(s []*spectypes.BeaconDuty, index int) []*spectypes.BeaconDuty {
-	ret := make([]*spectypes.BeaconDuty, 0)
-	ret = append(ret, s[:index]...)
-	return append(ret, s[index+1:]...)
+func removeIndices(s []*spectypes.BeaconDuty, indicesToRemove []int) []*spectypes.BeaconDuty {
+	sort.Sort(sort.Reverse(sort.IntSlice(indicesToRemove)))
+
+	for _, index := range indicesToRemove {
+		// Panic on incorrect indices
+		if index < 0 || index >= len(s) {
+			panic(fmt.Sprintf("index %d out of range of %v", index, indicesToRemove))
+		}
+		s = append(s[:index], s[index+1:]...)
+	}
+
+	return s
 }
```

Comment:
1. don't panic. - might be a security concern. return error, especially as this part of normal node operation.
2. It's better to not touch the original passed slice as its part of the `CommitteeRunner` and we "don't know" about its thread safety and such. create a new slice instead.